### DISCUSSION TITLE
[week 10] 綜合能力測驗-印出結果不是呈現中文

### DISCUSSION THREAD

**Discussion by chengcheng1231** [Lidemy/forum#159] [General]
想問一個問題以及分享一個已解決的事情。

**問題**：在玩 綜合能力測驗時 最後印出來的結果如果是呈現像 unicode 的形式應該是不正常的吧？如附圖，應該是要有「必須是 1 到 100 之間的數字」，看了一下 mattermost 大家好像沒這樣的問題 QQ

![image](https://user-images.githubusercontent.com/36998811/125159443-eef9c100-e1a9-11eb-9158-903ef4627aa4.png)

**分享**：同樣也是中文呈現的問題，本來拿到 ./script.js 回傳的結果中文會呈現亂碼，如下圖，但利用 [此網站 ](https://kusocloud.net/2017/04/24/%E8%A7%A3%E6%B1%BAchrome%E7%B6%B2%E9%A0%81%E7%B7%A8%E7%A2%BC%E9%A1%AF%E7%A4%BA%E4%BA%82%E7%A2%BC%E7%9A%84%E5%95%8F%E9%A1%8C/) 的方法，在 Google線上應用商店擴充 Set Character Encoding 並轉換成 UTF-8 就可以解決，而上述問題用此方式無法解決

![image](https://user-images.githubusercontent.com/36998811/125159595-fa99b780-e1aa-11eb-8ce5-fa24dae055d0.png)

**Comment by aszx87410** [comment]
應該是因為從 PHP 輸出就是那樣，其實算是正常，因為用 JS 來 parse 的話結果也會是中文
其他同學沒碰到應該是因為他們沒有獨立開這一頁，而是從遊戲的頁面開 devtool 去看，從那邊去看他就會自動幫你翻中文

**Reply by chengcheng1231** [reply]
謝謝 Huli，了解，現在知道還可以從 console 傳入遺漏的變數
![image](https://user-images.githubusercontent.com/36998811/125168142-7bba7400-e1d6-11eb-82ca-3aef3fa10c06.png)
